File: python/problem53.py
# refer:https://blog.csdn.net/lengxiao1993/article/details/52303492

class Solution:
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        ma=nums[0]
        m=nums[0]
        for i in nums[1::]:
            m=max(i,m+i)
            ma=max(ma,m)
        return ma

File: swift/problem53.swift
class Solution {
    func maxSubArray(_ nums: [Int]) -> Int {
        var sum = 0; var crest = Int.min
        
        for number in nums {
            sum += number
            crest = max(crest,sum)
            if (sum < 0) {
                sum = 0
            }
        }
        
        return crest
    }
}
